Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/ClusterManagerEx.java

```diff
@@ -135,6 +135,29 @@ public boolean isListAdditionalClusterSuccess() {
         }
     }
 
+    // Get HDInsight cluster detail from subscriptions
+    @Nullable
+    public ClusterDetail getHdiClusterDetailByName(@NotNull String clusterName) {
```

Comment:
What's difference than `getClusterDetailByName`?

Reply:
`getClusterDetailByName` will call `getConfigurationInfo` for each cluster, which is time-consuming and unnecessary in our scenario

---
In reply to: [278011824](https://github.com/Microsoft/azure-tools-for-java/pull/3075#discussion_r278011824) [](ancestors = 278011824)

Reply:
`getClusterDetailByName` will find the given cluster name in all kinds of clusters, including: 
     1. HDInsight clusters under user's azure subscription
     2. HDInsight linked clusters
     3. Emulator clusters
     4. SQL Big Data clusters
It might run into same cluster name between different cluster types issue when it search the given cluster name.
However, we only need to find a cluster with type `1. HDInsight clusters under user's azure subscription`

---
In reply to: [278011824](https://github.com/Microsoft/azure-tools-for-java/pull/3075#discussion_r278011824) [](ancestors = 278011824)